packages/dds/tree/src/simple-tree/api/snapshotCompatibilityChecker.ts

```diff
@@ -0,0 +1,112 @@
+import { toSimpleTreeSchema } from "./viewSchemaToSimpleSchema.js";
+
+/**
+ * Checks the compatibility of the view schema that created the document against the view schema being used to open it.
+ *
+ * @remarks See {@link SchemaCompatibilityStatus} for details on the compatibility results.
+ *
+ * @example This example demonstrates checking the compatibility of a historical schema against a current schema.
+ * In this case, the historical schema is a Point2D object with x and y fields, while the current schema is a Point3D object
+ * that adds an optional z field.
+ *
+ * ```ts
+ * // Build the current view schema
+ * const schemaFactory = new SchemaFactory("test");
+ * class Point3D extends schemaFactory.object("Point", {
+ *     x: factory.number,
+ *     y: factory.number,
+ *
+ *	   // The current schema has a new optional field that was not present on Point2D
+ *	   z: factory.optional(factory.number),
+ * }) {}
+ * const oldViewSchema = importCompatibilitySchemaSnapshot(fs.readFileSync("Point2D.json"));
+ *
+ * // Check to see if the document created by the historical view schema can be opened with the current view schema
+ * const compatibilityStatus = checkCompatibility(oldViewSchema, currentViewSchema);
+ *
+ * // z is not present in Point2D, so the schema must be upgraded
+ * assert.equal(compatibilityStatus.canView, false);
+ *
+ * // The schema can be upgraded to add the new optional field
+ * assert.equal(compatibilityStatus.canUpgrade, true);
+ * ```
+ *
+ * @param viewWhichCreatedDocument - The view schema which was used to create the document.
+ * @param view - The view schema being used to read the document.
+ * @returns The compatibility status.
+ *
+ * @alpha
+ */
+export function checkCompatibility(
+	viewWhichCreatedDocument: TreeViewConfiguration,
+	view: TreeViewConfiguration,
+): Omit<SchemaCompatibilityStatus, "canInitialize"> {
+	const viewAsAlpha = new TreeViewConfigurationAlpha({ schema: view.schema });
+	const stored = toStoredSchema(viewWhichCreatedDocument.schema, {
+		includeStaged: () => true,
+	});
+	const tester = new SchemaCompatibilityTester(viewAsAlpha);
+	return tester.checkCompatibility(stored);
+}
+
+/**
+ * Returns a JSON representation of the tree schema for snapshot compatibility checking.
+ * @param config - The schema to snapshot. Only the schema field of the `TreeViewConfiguration` is used.
+ * @returns The JSON representation of the schema.
+ *
+ * @example This example creates and persists a snapshot of a Point2D schema.
+ *
+ * ```ts
+ * const schemaFactory = new SchemaFactory("test");
+ * class Point2D extends schemaFactory.object("Point", {
+ *     x: factory.number,
+ *     y: factory.number,
+ * }) {}
+ * const viewSchema = new TreeViewConfiguration({ schema: Point2D });
+ * fs.writeFileSync("Point2D.json", exportCompatibilitySchemaSnapshot(viewSchema));
+ * ```
+ *
+ * @alpha
+ */
+export function exportCompatibilitySchemaSnapshot(
```

Comment:
Neither the docs or example link importCompatibilitySchemaSnapshot which really should be linked as this function is useless without it.